src/cmd-buildextend-azure

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+# pylint: disable=C0103
+"""
+An operation that mutates a build by uploading to Azure,
+extending the meta.json with the Azure image name.
+"""
+# NOTE: PYTHONUNBUFFERED is set in cmdlib.sh for unbuffered output
+
+import argparse
+import logging as log
+import os
+import sys
+
+cosa_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, cosa_dir)
+
+try:
+    from cosalib.build import _Build
+except ImportError:
+    # We're in the container and can't sense the cosa path
+    sys.path.insert(0, '/usr/lib/coreos-assembler/cosalib')
+    from cosalib.build import _Build
+
+# pylint: disable=C0413
+from cmdlib import (
+        run_verbose,
+        sha256sum_file,
+        write_json)
+
+# Identifier for azure platform
+PLATFORM_AZURE = 'azure'
+
+
+class Build(_Build):
+    """
+    Azure implementation of Build.
+    """
+
+    def _build_artifacts(self, *args, **kwargs):
+        """
+        Implements the building of artifacts. Walk the build root and
+        prepare a list of files in it.
+        :param args: All non-keyword arguments
+        :type args: list
+        :param kwargs: All keyword arguments
+        :type kwargs: dict
+        """
+        self._create_tmp_dir()
+
+        # Name the base build and tarball name
+        base_name = self.meta['name']
+        azure_nv = f'{base_name}-{self.build_id}'
+
+        # Used in referencing
+        self.azure_vhd_name = f'{azure_nv}.vhd'
+
+        # Generate path locations
+        img_qemu = os.path.join(
+            self.build_root, self.meta['images']['qemu']['path'])
+
+        tmp_img_azure = os.path.join(self.workdir, (azure_nv + '.qcow2'))
```

Comment:
Aside: I think we should probably change `workdir` to e.g. `tmpbuilddir` or something? `workdir` already means something else in the rest of the codebase.

Reply:
Will do as a followup